core/state/statedb.go

```diff
@@ -204,6 +204,9 @@ func (s *StateDB) Error() error {
 func (s *StateDB) AddLog(log *types.Log) {
 	s.journal.append(addLogChange{txhash: s.thash})
```

Comment:
lock should be before this line, also there's a function GetLogs() and Logs() right be low this one, should it acquire the read lock?